### DISCUSSION TITLE
terrascan integration to Dagger

### DISCUSSION THREAD

**Discussion by sangam14** [dagger/dagger#1882] [Ideas]
https://github.com/accurics/terrascan

Detect compliance and security violations across Infrastructure as Code to mitigate risk before provisioning cloud native infrastructure.

500+ Policies for security best practices
Scanning of Terraform (HCL2)
Scanning of Kubernetes (JSON/YAML), Helm v3, and Kustomize v3
Support for AWS, Azure, GCP, Kubernetes and GitHub

Idea to scan dagger application before deploying  

we are planning to add cue support (https://github.com/accurics/terrascan/issues/896)

**Comment by aluzzardi** [comment]
@sangam14 Do you think this should be implemented as a package, e.g. `terrascan.#Scan`, so it can be used in a pipeline (e.g. scan kube yamls with terrascan, then apply them)?

Or were you envisioning a different integration?

**Comment by sangam14** [comment]
as package make more robust within pipeline  `import "alpha.dagger.io/terrascan "` . dagger cue package already support yaml .
